beacon/goclient/goclient.go

```diff
@@ -150,6 +153,9 @@ type GoClient struct {
 	nodeClient  NodeClient
 	gasLimit    uint64
 
+	allowUnsyncedSlots phase0.Slot
```

Comment:
Can I propose the name `syncDistance` or `syncTolerance`? @anatolie-ssv @moshe-blox @MatusKysel